src/main/java/com/cloudant/search3/Search.java

```diff
@@ -174,9 +173,9 @@ public void onRemoval(final RemovalNotification<SearchCacheKey, SearchHandler> n
             try {
                 commitFutures.remove(notification.getKey()).cancel(false);
                 notification.getValue().close();
-                LOGGER.info("Closed handler for index {}.",  notification.getKey());
+                LOGGER.info("Closed handler for index {} for reason {}.",  notification.getValue(), notification.getCause());
             } catch (final IOException e) {
-                LOGGER.error("I/O exception while closing evicted index " + notification.getKey(), e);
+                LOGGER.error("I/O exception while closing evicted index " + notification.getValue(), e);
```

Comment:
The log4j string formatting support is pretty robust I have to say, although it led to early confusion on my part.

https://logging.apache.org/log4j/2.x/manual/messages.html

Reply:
it's annoying that you (apparently) can't do a formatted message and include an exception, a limitation of varargs approach,.